### PR TITLE
CRIMRE-142 Add sorting buttons (WIP)

### PR DESCRIPTION
**Pull request by timpeat**
Co-authored-by: willmcb <william.mcbrien@digital.justice.gov.uk>

## Description of change

Adds buttons and hooks up controllers for sorting search results.

NB this is pending styling and api client work.

## Link to relevant ticket
[CRIMRE-45](https://dsdmoj.atlassian.net/browse/CRIMRE-45)

### After changes:

Screenshot 2023-01-26 at 17 54 59

## How to manually test the feature

[CRIMRE-45]: https://dsdmoj.atlassian.net/browse/CRIMRE-45?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/models/sorting.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Sorting < ApplicationStruct
+  attribute? :sort_direction, Types::String.default('descending'.freeze).enum('descending', 'ascending')
```
Comment (willmcb) [inline app/models/sorting.rb:2]: Add these to types

File: app/models/sorting.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+             Types::String.default('submitted_at'.freeze).enum('submitted_at', 'reviewed_at', 'reference',
+                                                               'applicant_name')
+
+  def reverse_direction
+    return 'ascending' if sort_direction == 'descending'
+
+    'descending'
+  end
+
+  def aria_sort(column_name)
```
Comment (willmcb) [inline app/models/sorting.rb:13]: Change name to `column_sort_state`

File: spec/system/searching/filter_by_caseworker_spec.rb [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
   context 'when unassigned status is selected' do
     before do
-      select 'Unassigned', from: 'filter-assigned-status-field'
+      select 'Unassigned', from: 'search-application-search-filter-assigned-status-field'
```
Comment (willmcb) [inline spec/system/searching/filter_by_caseworker_spec.rb:19]: Perhaps add as a let to DRY out

File: spec/system/searching/filter_by_date_of_birth_spec.rb [hunk 4]
```diff
@@ -8,7 +8,7 @@
     visit '/'
 
     click_on 'Search'
-    fill_in 'filter-applicant-date-of-birth-field', with: dob
+    fill_in 'search-application-search-filter-applicant-date-of-birth-field', with: dob
```
Comment (willmcb) [inline spec/system/searching/filter_by_date_of_birth_spec.rb:11]: ditto for this string

**Review by willmcb — APPROVED** [review]
Nice 👍

**Review by timpeat — COMMENTED** [review]
Updated as per review comments.

**Pull request merged**